spec/requests/hyku_addons/multitenant_locale_controller_behavior_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe HykuAddons::MultitenantLocaleControllerBehavior, type: :request, multitenant: true do
+  let(:main_app) { Rails.application.routes.url_helpers }
+  let(:user) { create(:user) }
+  let(:account) { create(:account) }
+  let(:work) { create(:work, user: user) }
+
+  before do
+    login_as(user, scope: :user)
+
+    Site.update(account: account)
+    allow(Apartment::Tenant).to receive(:switch).with(account.tenant) do |&block|
+      block.call
+    end
+    host! account.cname
+  end
+
+  describe "when no locale name is set on an account" do
+    before do
+      account.settings["locale_name"] = nil
+      account.save
+
+      get main_app.polymorphic_path(work)
+    end
+
+    it "returns a normal locale code" do
+      expect(response.body).to include('lang="en"')
+    end
+  end
+
+  describe "when a locale name is set on an account" do
+    before do
+      account.settings["locale_name"] = "test"
+      account.save
+
+      get main_app.polymorphic_path(work)
+    end
+
+    it "appends the locale name to the languge code" do
+      expect(response.body).to include('lang="en-TEST"')
```

Comment:
Where is this in the body?  I would hope that `en-TEST` isn't displayed to the user at all: location url or page body.  Instead I would hope the user only ever sees `en`.

Reply:
It's not visible to the user, but part of the markup. The user only ever sees the locale in the URL (if they look there). This seemed a simple way to confirm that the locale was loaded. However, if you know if a better way, i'm more than happy to change it.